samples/sample_plugins/rotate_opencl/src/opencl_filter.cpp

```diff
@@ -127,43 +128,104 @@ cl_int OpenCLFilterBase::InitPlatform()
         return error;
     }
 
     // Get all of the handles to the installed OpenCL platforms
     std::vector<cl_platform_id> platforms(num_platforms);
     error = clGetPlatformIDs(num_platforms, &platforms[0], &num_platforms);
-    if(error) {
-        log.error() << "OpenCLFilter: Failed to get OCL platform IDs. Error Code; " << error;
+    if(error)
+    {
+        log.error() << "OpenCLFilter: Failed to get OCL platform IDs."
+                    << " Error code: " << error << endl;
         return error;
     }
 
-    // Find the platform handle for the installed Gen driver
-    const size_t max_string_size = 1024;
-    char platform[max_string_size];
-    cl_device_id device_ids[2] = {0};
-    for (unsigned int platform_index = 0; platform_index < num_platforms; platform_index++)
+    for (std::vector<cl_platform_id>::iterator platformIt = platforms.begin(); platformIt != platforms.end(); ++platformIt)
     {
-        error = clGetPlatformInfo(platforms[platform_index], CL_PLATFORM_NAME, max_string_size, platform, NULL);
-        if(error) return error;
+        cl_uint num_devices = 0;
 
-        // Choose only GPU devices
-        if (clGetDeviceIDs(platforms[platform_index], CL_DEVICE_TYPE_GPU,
-            sizeof(device_ids)/sizeof(device_ids[0]), device_ids, 0) != CL_SUCCESS)
+        error = clGetDeviceIDs(*platformIt, CL_DEVICE_TYPE_GPU, 0, NULL, &num_devices);
+        if(error)
+        {
+            log.warning() << "OpenCLFilter: Couldn't get number of GPU devices for current platform."
+                          << " Error code: " << error << endl;
             continue;
+        }
+
+        std::vector<cl_device_id> devices(num_devices);
 
-        if(strstr(platform, "Intel")) // Use only Intel platfroms
+        error = clGetDeviceIDs(*platformIt, CL_DEVICE_TYPE_GPU, devices.size(), &devices[0], 0);
+        if(error)
         {
-            log.info() << "OpenCL platform \"" << platform << "\" is used" << endl;
-            m_clplatform = platforms[platform_index];
+            log.error() << "OpenCLFilter: Couldn't get GPU device IDs."
+                        << " Error code: " << error << endl;
+            continue;
+        }
+
+        for(std::vector<cl_device_id>::iterator deviceIt = devices.begin(); deviceIt != devices.end(); ++deviceIt)
+        {
+            cl_uint deviceVendorId = 0;
+
+            error = clGetDeviceInfo(*deviceIt, CL_DEVICE_VENDOR_ID, sizeof(deviceVendorId), &deviceVendorId, NULL);
+
+            if (error)
+            {
+                log.error() << "OpenCLFilter: Couldn't get the device vendor id."
+                    << " Error code: " << error << endl;
+                continue;
+            }
+
+            // Skip non-Intel devices
+            if (deviceVendorId != 0x8086)
+            {
+                continue;
+            }
+
+            size_t extNamesBufferSize = 0;
```

Comment:
I would rename these variables to something more simple. `extensionsNum` and `extensions` would be better

Reply:
Renamed.